From 128a99decdfd697f6426e3ee651e852081dd7ac5 Mon Sep 17 00:00:00 2001
From: limeidan <limeidan@loongson.cn>
Date: Fri, 1 Mar 2024 11:13:21 +0800
Subject: [PATCH] cmd/link/internal/loong64: correct the musl dynamic linker
 path

Musl libc already supports loongarch64, the dynamic linker path is the
same as other architectures: /lib/ld-musl-$ARCH.so.1

Ref: https://git.musl-libc.org/cgit/musl/tree/INSTALL#n141

Change-Id: Ie31d4254f6e14a5d634b2a7b5fa4c6270e2c0dc7
Reviewed-on: https://go-review.googlesource.com/c/go/+/569475
Reviewed-by: Cherry Mui <cherryyz@google.com>
Auto-Submit: Cherry Mui <cherryyz@google.com>
Reviewed-by: Than McIntosh <thanm@google.com>
LUCI-TryBot-Result: Go LUCI <golang-scoped@luci-project-accounts.iam.gserviceaccount.com>
Reviewed-by: abner chenc <chenguoqi@loongson.cn>
Reviewed-by: sophie zhao <zhaoxiaolin@loongson.cn>
Reviewed-by: Qiqi Huang <huangqiqi@loongson.cn>
---
 src/cmd/link/internal/loong64/obj.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cmd/link/internal/loong64/obj.go b/src/cmd/link/internal/loong64/obj.go
index b68ed494f6..06ac5302cc 100644
--- a/src/cmd/link/internal/loong64/obj.go
+++ b/src/cmd/link/internal/loong64/obj.go
@@ -30,7 +30,7 @@ func Init() (*sys.Arch, ld.Arch) {
 
 		ELF: ld.ELFArch{
 			Linuxdynld:     "/lib64/ld-linux-loongarch-lp64d.so.1",
-			LinuxdynldMusl: "/lib64/ld-musl-loongarch.so.1",
+			LinuxdynldMusl: "/lib/ld-musl-loongarch.so.1",
 			Freebsddynld:   "XXX",
 			Openbsddynld:   "XXX",
 			Netbsddynld:    "XXX",
-- 
2.45.1

